engine/include/cubos/engine/assets/asset.hpp

```diff
@@ -12,6 +12,14 @@
 
 namespace cubos::engine
 {
+    /// @brief Type of the asset identifier.
+    enum class IdType
+    {
+        UUID,
+        Path,
+        Invalid
+    };
```

Comment:
To avoid polluting the whole `cubos::engine` namespace, we usually prefix these types with the relevant keywords (e.g. `Asset`) or move them within the class itself, so that it becomes something like `AnyAsset::IdType`.